pkg/cloudprovider/fake/cloudprovider.go

```diff
@@ -52,17 +54,26 @@ func NewCloudProvider() *CloudProvider {
 	}
 }
 
-func (c *CloudProvider) Create(ctx context.Context, nodeRequest *cloudprovider.NodeRequest) (*v1.Node, error) {
+func (c *CloudProvider) Create(ctx context.Context, machine *v1alpha1.Machine) (*v1.Node, error) {
 	c.mu.Lock()
-	c.CreateCalls = append(c.CreateCalls, nodeRequest)
+	c.CreateCalls = append(c.CreateCalls, machine)
 	if len(c.CreateCalls) > c.AllowedCreateCalls {
 		c.mu.Unlock()
 		return &v1.Node{}, fmt.Errorf("erroring as number of AllowedCreateCalls has been exceeded")
 	}
 	c.mu.Unlock()
 
-	name := test.RandomName()
-	instanceType := nodeRequest.InstanceTypeOptions[0]
+	requirements := scheduling.NewNodeSelectorRequirements(machine.Spec.Requirements...)
```

Comment:
Do you need this if you have machine.Requirements()? Probably safer to not have v1alpha1 rely on the scheduling package.